### PR TITLE
Remove more code

### PR DESCRIPTION
**Pull request by olekenneth**
I don't like else statements (better with sane defaults)
The protocol check is not needed anymore with the isUrl-function
Also not needed to declare redirect on global scope and then call

### CODE REVIEW

File: 404.html [hunk 1]
```diff
@@ -48,30 +50,24 @@
...
-                  throw new Error("URL to redirect to is invalid!");
+                // Prevent recursive redirects 
+                if (url.host !== location.host) {
+                  location.replace(title);
                 }
-
-                location.replace(title);
-              } else {
-                throw new Error("issueNumber does not exist in gh issues");
+                throw new Error("URL to redirect to is invalid!");
```
Comment (nelsontky) [inline 404.html:61]: This statement will run everytime, which is not desired behavior

File: 404.html [hunk 2]
```diff
@@ -48,30 +50,24 @@
...
               }
+              throw new Error("issueNumber does not exist in gh issues");
```
Comment (nelsontky) [inline 404.html:63]: Same goes with this line, and thus the `else` exists. To remove the `else`, just invert the condition, and throw errors in the `if` block

File: 404.html [hunk 3]
```diff
@@ -48,30 +50,24 @@
...
             });
           })
           .catch(function (e) {
```
Comment (nelsontky) [inline 404.html:66]: We could remove `e` and the subsequent comment

File: 404.html [hunk 4]
```diff
@@ -48,30 +50,24 @@
...
+            // alert(e.message);
             location.replace(homepage);
           });
-      }
-
-      redirect();
+      })();
```
Comment (nelsontky) [inline 404.html:70]: I personally see no issue with declaring functions in the global scope. And I don't really like calling a declared function immediately that's just a personal thing HAHA

File: 404.html (hunk 1 above)
Comment (olekenneth) [reply 404.html:59]: Ahh! Then we need a return;

File: 404.html (hunk 4 above)
Comment (olekenneth) [reply 404.html]: Sure. You're the owner ;-)

**Pull request merged**